src/CharacterList.js

```diff
@@ -8,46 +8,79 @@ class CharacterList extends Component {
     super(props);
     this.state = {
       characters: [],
-      letter: this.props.params.letter || "a"
+      total: 0,
+      letter: this.props.params.letter || "a",
+      page: 0,
+      paging: this.props.paging || 20,
+      count: 0,
+      offset: 0
     };
     this.fetchCharacters = this.fetchCharacters.bind(this);
+    this.resetPage = this.resetPage.bind(this);
+    this.incrementPage = this.incrementPage.bind(this);
+    this.decrementPage = this.decrementPage.bind(this);
   }
 
   componentDidMount() {
-    this.fetchCharacters(this.state.letter);
+    this.fetchCharacters(this.state.letter, 0);
```

Comment:
it needs to start at 0 if we are reloading the page, so I am going to leave it like this